VERSIONING.md

```diff
@@ -30,9 +30,40 @@ Kotlin version | Minimal Compose version | Notes
 1.5.31 | 1.0.0
 1.6.20 | 1.1.1
 1.7.10 | 1.2.0
-1.7.20 | 1.2.0 | JS is not supported (will be fixed in the next versions)
+1.7.20 | 1.2.0 | JS is not supported (fixed in the next version)
 1.7.20 | 1.2.1
 
+### Using the latest Kotlin version 
+
+When a new version of Kotlin is released, the corresponding Compose Multiplatform release may not yet have been published. There are still ways to use it, although stability is not garantied.
+
+#### Using Jetpack Compose Compiler
+
+Compose Multiplatform uses Compose Compiler to compile Compose code, and its version is strictly bound to the version of Kotlin. By default it chooses appropriate `org.jetbrains.compose.compiler:compiler` version by the Kotlin version applied to the Gradle project. But there is a way to choose another version of Compose Compiler. For example, you can use Jetpack Compose Compiler published by Google
+
+First check [this page](https://developer.android.com/jetpack/androidx/releases/compose-kotlin#pre-release_kotlin_compatibility) to find a compatible version. If there is one, use it this way:
+```
+compose {
+    kotlinCompilerPlugin.set("androidx.compose.compiler:compiler:1.4.0-alpha02")
+}
+```
+(`1.4.0-alpha02` corresponds Kotlin 1.7.21)
+
+#### Disabling Kotlin compatability check
+
+If there is no compatible version of Jetpack Compose Compiler, or you encounter errors, you can try to use Compose Compiler for another version of Kotlin, but disable the Kotlin version check:
+
+```
+compose {
+    kotlinCompilerPlugin.set(dependencies.compiler.forKotlin("1.7.20"))
+    kotlinCompilerPluginArgs.add("suppressKotlinVersionCompatibilityCheck=1.7.21")
```

Comment:
The reason why I added support for arbitrary properties - because there are other useful properties besides `suppressKotlinCompatibilityCheck`, and I wouldn't want to modify code every time new properties/rules are added. For example, the useful [metricsDestination](https://chris.banes.dev/posts/composable-metrics/) property was added recently.

Keeping API under control is usually good practice, and allows avoiding many issues, but here we explicitly tell that we use a separate entity (Compose Compiler), and so we can support configuration of it bypassing our API, and reusing its own way of configuration (via properties).

Besides that, we should emphasis that we configure the applied Compose Compiler. Without specifying a custom Compose Compiler, arguments don't make sense, because Compiler can be different for each Kotlin and doesn't have required arguments. In case `suppressKotlinCompatibilityCheck` we have another problem - it won't be used until we set a custom compiler.

So, if we really don't want to pass arbitrary properties (but I still prefer to pass because it is easier to maintain), we should keep somehow connection to Compose Compiler. The alternative are:

```
compose {
    kotlinCompilerPlugin(dependencies.compiler.forKotlin("1.7.20")) {
        suppressKotlinCompatibilityCheck = "1.7.21"
    }
}

// and we can still write
compose {
    kotlinCompilerPlugin("androidx.compose.compiler:compiler:1.4.0-alpha02")
}

```
(I am not sure about the style guidelines, maybe it can be written better)

or
```
compose {
    kotlinCompilerPlugin.set(dependencies.compiler.forKotlin("1.7.20"))
    kotlinCompilerPluginArgs {
        suppressKotlinCompatibilityCheck.set("1.7.21")
    }
}
```

Reply:
> The alternative are

It seems that they won't work good. For example, we have the property `metricsDestination`. It only makes sense for the latest Compose Compiler, and will be ignored in the previous versions.